code/ui/blocks/src/blocks/Source.tsx

```diff
@@ -94,50 +102,50 @@ export const useSourceProps = (
   docsContext: DocsContextProps<any>,
   sourceContext: SourceContextProps
 ): PureSourceProps & SourceStateProps => {
-  const { id: primaryId, parameters } = docsContext.storyById();
-
-  const codeProps = props as CodeProps;
-  const singleProps = props as SingleSourceProps;
-  const multiProps = props as MultiSourceProps;
-
-  let { format, code: source } = codeProps; // prefer user-specified code
-
-  const targetIds = multiProps.ids || [singleProps.id || primaryId];
-  const storyIds = targetIds.map((targetId) => {
-    return targetId;
-  });
-
-  const stories = useStories(storyIds, docsContext);
-  if (!stories.every(Boolean)) {
+  const storyIds = props.ids || (props.id ? [props.id] : []);
+  const storiesFromIds = useStories(storyIds, docsContext);
+  if (!storiesFromIds.every(Boolean)) {
     return { error: SourceError.SOURCE_UNAVAILABLE, state: SourceState.NONE };
   }
 
-  if (!source) {
-    // just take the format from the first story, given how they're all concatinated together...
-    // TODO: we should consider sending an event with all the sources separately, instead of concatenating them here
-    ({ format } = getStorySource(storyIds[0], sourceContext));
-    source = storyIds
-      .map((storyId, idx) => {
-        const { code: storySource } = getStorySource(storyId, sourceContext);
-        const storyObj = stories[idx] as PreparedStory;
-        return getSnippet(storySource, storyObj);
+  // The check didn't actually change the type.
+  let stories: PreparedStory[] = storiesFromIds as PreparedStory[];
+  if (props.of) {
+    const resolved = docsContext.resolveOf(props.of, ['story']);
+    stories = [resolved.story];
+  } else if (stories.length === 0) {
+    stories = [docsContext.storyById()];
+  }
+
+  const sourceParameters = (stories[0].parameters.docs?.source || {}) as SourceParameters;
+  let { code } = props; // We will fall back to `sourceParameters.code`, but per story below
+  let format = props.format ?? sourceParameters.format;
+  const language = props.language ?? sourceParameters.language ?? 'jsx';
+  const dark = props.dark ?? sourceParameters.dark ?? false;
+
+  if (!code) {
+    code = stories
+      .map((story, index) => {
```

Comment:
Specifically this conditional seems to be the source of the problem I'm seeing. Because in my case I specifically set the code property on all my stories, this conditional is bypassing the story source calculation. I'm not sure if there are other issues downstream, but I'm pretty sure until this conditional is somehow changed I won't get the right source displayed.